contracts/PrepaidCardManager.sol

```diff
@@ -212,20 +220,34 @@ contract PrepaidCardManager is Ownable, Versionable, Safe {
       "the amount arrays have differing lengths"
     );
 
-    // The spend amounts are for reporting purposes only, there is no on-chain
-    // effect from this value. Although, it might not be a bad idea that spend
-    // amounts line up with the issuing token amounts--albiet we'd need to
-    // introduce a rate lock mechanism if we wanted to validate this
-    createMultiplePrepaidCards(
-      owner,
-      from,
-      _msgSender(),
-      amount,
-      issuingTokenAmounts,
-      spendAmounts,
-      customizationDID,
-      marketAddress
-    );
+    if (issuer == address(0)) {
+      // The spend amounts are for reporting purposes only, there is no on-chain
+      // effect from this value. Although, it might not be a bad idea that spend
+      // amounts line up with the issuing token amounts--albiet we'd need to
+      // introduce a rate lock mechanism if we wanted to validate this
+      createMultiplePrepaidCards(
+        owner,
+        from,
+        _msgSender(),
+        amount,
+        issuingTokenAmounts,
+        spendAmounts,
+        customizationDID,
+        marketAddress
+      );
+    } else {
+      createCardWithIssuer(
```

Comment:
This is a trusted operation--since we need to trust that the caller is forthright in terms of who the issuer is. We should add a require that only the PrepaidCardMarketV2 contract is able to call this (the `from` address). Which also means that the `setup` function needs to be provided with an array of trusted callers for this type of prepaidcard creation (it should be an array to be compatible with future protocol contracts that may want to do this).